### ISSUE TITLE
Build failing because npm audit need fix

### ISSUE THREAD

**Issue by merlinofcha0s** [jhipster/generator-jhipster#14934]
##### **Overview of the issue**
It seems that the build is failing with this stage `npm run java:jar:prod` when there are some audit to fix during `npm install`
it's failing on the CI also at this stage : 

<img width="643" alt="image" src="https://user-images.githubusercontent.com/6643835/117753887-9180dc80-b219-11eb-9eac-161d21b394d1.png">

It's very recent issue, yesterday it was okay.

Also the problematic dependency is this one `"@angular-builders/custom-webpack": "11.1.1"` : 

<img width="663" alt="image" src="https://user-images.githubusercontent.com/6643835/117754492-b75ab100-b21a-11eb-8d72-59eb666c4192.png">

Even when upgrading to the next beta version the dependency is still not okay.
 
See the stacktrace :

`> lol@0.0.1-SNAPSHOT java:jar:prod /Users/raiden/IdeaProjects/lol
> npm run java:jar -- -Pprod

> lol@0.0.1-SNAPSHOT java:jar /Users/raiden/IdeaProjects/lol
> ./mvnw -ntp verify -DskipTests --batch-mode "-Pprod"

[INFO] Scanning for projects...
[INFO] 
[INFO] ---------------------------< fr.lol:lol >---------------------------
[INFO] Building lol 0.0.1-SNAPSHOT
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] 
[INFO] --- maven-resources-plugin:3.2.0:copy-resources (default-resources) @ lol ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] Using 'UTF-8' encoding to copy filtered properties files.
[INFO] Copying 4 resources
[INFO] Copying 29 resources
[INFO] 
[INFO] --- maven-resources-plugin:3.2.0:resources (default-resources) @ lol ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] Using 'UTF-8' encoding to copy filtered properties files.
[INFO] Copying 4 resources
[INFO] Copying 29 resources
[INFO] 
[INFO] --- maven-enforcer-plugin:3.0.0-M3:enforce (enforce-versions) @ lol ---
[INFO] 
[INFO] --- maven-enforcer-plugin:3.0.0-M3:enforce (enforce-dependencyConvergence) @ lol ---
[WARNING] 
Dependency convergence error for net.java.dev.jna:jna:5.2.0 paths to dependency are:
+-fr.lol:lol:0.0.1-SNAPSHOT
  +-org.testcontainers:postgresql:1.15.2
    +-org.testcontainers:jdbc:1.15.2
      +-org.testcontainers:database-commons:1.15.2
        +-org.testcontainers:testcontainers:1.15.2
          +-org.rnorth.visible-assertions:visible-assertions:2.1.2
            +-net.java.dev.jna:jna:5.2.0
and
+-fr.lol:lol:0.0.1-SNAPSHOT
  +-org.testcontainers:postgresql:1.15.2
    +-org.testcontainers:jdbc:1.15.2
      +-org.testcontainers:database-commons:1.15.2
        +-org.testcontainers:testcontainers:1.15.2
          +-com.github.docker-java:docker-java-transport-zerodep:3.2.7
            +-net.java.dev.jna:jna:5.5.0

[WARNING] 
Dependency convergence error for org.wildfly.common:wildfly-common:1.5.2.Final paths to dependency are:
+-fr.lol:lol:0.0.1-SNAPSHOT
  +-org.springframework.boot:spring-boot-starter-undertow:2.4.4
    +-io.undertow:undertow-core:2.2.5.Final
      +-org.jboss.xnio:xnio-api:3.8.0.Final
        +-org.wildfly.common:wildfly-common:1.5.2.Final
and
+-fr.lol:lol:0.0.1-SNAPSHOT
  +-org.springframework.boot:spring-boot-starter-undertow:2.4.4
    +-io.undertow:undertow-core:2.2.5.Final
      +-org.jboss.xnio:xnio-api:3.8.0.Final
        +-org.wildfly.client:wildfly-client-config:1.0.1.Final
          +-org.wildfly.common:wildfly-common:1.2.0.Final

[WARNING] 
Dependency convergence error for org.apiguardian:apiguardian-api:1.1.0 paths to dependency are:
+-fr.lol:lol:0.0.1-SNAPSHOT
  +-org.springframework.boot:spring-boot-starter-test:2.4.4
    +-org.junit.jupiter:junit-jupiter:5.7.1
      +-org.junit.jupiter:junit-jupiter-api:5.7.1
        +-org.apiguardian:apiguardian-api:1.1.0
and
+-fr.lol:lol:0.0.1-SNAPSHOT
  +-org.springframework.boot:spring-boot-starter-test:2.4.4
    +-org.junit.jupiter:junit-jupiter:5.7.1
      +-org.junit.jupiter:junit-jupiter-api:5.7.1
        +-org.junit.platform:junit-platform-commons:1.7.1
          +-org.apiguardian:apiguardian-api:1.1.0
and
+-fr.lol:lol:0.0.1-SNAPSHOT
  +-org.springframework.boot:spring-boot-starter-test:2.4.4
    +-org.junit.jupiter:junit-jupiter:5.7.1
      +-org.junit.jupiter:junit-jupiter-params:5.7.1
        +-org.apiguardian:apiguardian-api:1.1.0
and
+-fr.lol:lol:0.0.1-SNAPSHOT
  +-org.springframework.boot:spring-boot-starter-test:2.4.4
    +-org.junit.jupiter:junit-jupiter:5.7.1
      +-org.junit.jupiter:junit-jupiter-engine:5.7.1
        +-org.apiguardian:apiguardian-api:1.1.0
and
+-fr.lol:lol:0.0.1-SNAPSHOT
  +-com.tngtech.archunit:archunit-junit5-engine:0.17.0
    +-com.tngtech.archunit:archunit-junit5-engine-api:0.17.0
      +-org.junit.platform:junit-platform-engine:1.7.1
        +-org.apiguardian:apiguardian-api:1.1.0
and
+-fr.lol:lol:0.0.1-SNAPSHOT
  +-org.zalando:problem-spring-web:0.26.2
    +-org.zalando:problem-violations:0.26.2
      +-org.apiguardian:apiguardian-api:1.1.0
and
+-fr.lol:lol:0.0.1-SNAPSHOT
  +-org.zalando:problem-spring-web:0.26.2
    +-org.zalando:problem-spring-common:0.26.2
      +-org.apiguardian:apiguardian-api:1.1.0
and
+-fr.lol:lol:0.0.1-SNAPSHOT
  +-org.zalando:problem-spring-web:0.26.2
    +-org.apiguardian:apiguardian-api:1.1.0
and
+-fr.lol:lol:0.0.1-SNAPSHOT
  +-org.zalando:problem-spring-web:0.26.2
    +-org.zalando:faux-pas:0.8.0
      +-org.apiguardian:apiguardian-api:1.0.0

[WARNING] Rule 0: org.apache.maven.plugins.enforcer.DependencyConvergence failed with message:
Failed while enforcing releasability. See above detailed error message.
[INFO] 
[INFO] --- jacoco-maven-plugin:0.8.6:prepare-agent (pre-unit-tests) @ lol ---
[INFO] argLine set to -javaagent:/Users/raiden/.m2/repository/org/jacoco/org.jacoco.agent/0.8.6/org.jacoco.agent-0.8.6-runtime.jar=destfile=/Users/raiden/IdeaProjects/lol/target/jacoco.exec -Djava.security.egd=file:/dev/./urandom -Xmx256m
[INFO] 
[INFO] --- properties-maven-plugin:1.0.0:read-project-properties (default) @ lol ---
[INFO] 
[INFO] --- git-commit-id-plugin:4.0.4:revision (default) @ lol ---
[INFO] 
[INFO] --- spring-boot-maven-plugin:2.4.4:build-info (default) @ lol ---
[INFO] 
[INFO] --- frontend-maven-plugin:1.11.3:install-node-and-npm (install node and npm) @ lol ---
[INFO] Node v14.16.0 is already installed.
[INFO] NPM 7.12.0 is already installed.
[INFO] 
[INFO] --- frontend-maven-plugin:1.11.3:npm (npm install) @ lol ---
[INFO] Running 'npm install' in /Users/raiden/IdeaProjects/lol
[INFO] 
[INFO] up to date, audited 2549 packages in 5s
[INFO] 
[INFO] 4 moderate severity vulnerabilities
[INFO] 
[INFO] To address all issues (including breaking changes), run:
[INFO]   npm audit fix --force
[INFO] 
[INFO] Run `npm audit` for details.
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  15.918 s
[INFO] Finished at: 2021-05-11T11:11:30+08:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal com.github.eirslett:frontend-maven-plugin:1.11.3:npm (npm install) on project lol: Failed to run task: 'npm install' failed. org.apache.commons.exec.ExecuteException: Process exited with an error: 1 (Exit value: 1) -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! lol@0.0.1-SNAPSHOT java:jar: `./mvnw -ntp verify -DskipTests --batch-mode "-Pprod"`
npm ERR! Exit status 1
`

##### **Motivation for or Use Case**

My build is failing everywhere so it's important

##### **Reproduce the error**

Generate a project and launch `npm run java:jar:prod`

##### **JHipster Version(s)**

7.0.1

##### **JHipster configuration**

Welcome to the JHipster Information Sub-Generator

##### **JHipster Version(s)**

```
lol@0.0.1-SNAPSHOT /Users/raiden/IdeaProjects/lol
└── generator-jhipster@7.0.1 

```

##### **JHipster configuration, a `.yo-rc.json` file generated in the root folder**

<details>
<summary>.yo-rc.json file</summary>
<pre>
{
  "generator-jhipster": {
    "blueprints": [],
    "otherModules": [],
    "applicationType": "monolith",
    "baseName": "lol",
    "jhipsterVersion": "7.0.1",
    "skipClient": false,
    "skipServer": false,
    "skipUserManagement": true,
    "skipCheckLengthOfIdentifier": false,
    "skipFakeData": false,
    "jhiPrefix": "jhi",
    "entitySuffix": "",
    "dtoSuffix": "DTO",
    "testFrameworks": ["cypress"],
    "pages": [],
    "creationTimestamp": 1617866281095,
    "serviceDiscoveryType": false,
    "reactive": false,
    "authenticationType": "oauth2",
    "packageName": "fr.lol",
    "serverPort": "8080",
    "cacheProvider": "hazelcast",
    "enableHibernateCache": true,
    "databaseType": "sql",
    "devDatabaseType": "h2Disk",
    "prodDatabaseType": "postgresql",
    "buildTool": "maven",
    "serverSideOptions": [],
    "websocket": false,
    "searchEngine": false,
    "messageBroker": false,
    "enableSwaggerCodegen": false,
    "clientFramework": "angularX",
    "withAdminUi": true,
    "clientTheme": "litera",
    "clientThemeVariant": "primary",
    "enableTranslation": true,
    "nativeLanguage": "fr",
    "packageFolder": "fr/lol",
    "clientPackageManager": "npm",
    "languages": ["fr", "en"],
    "entities": ["Profile", "Company", "CompanyStats", "EmployeeAsset", "GroupAsset", "GroupAssetRef"],
    "lastLiquibaseTimestamp": 1619752484000
  }
}

</pre>
</details>

##### **Environment and Tools**

openjdk version "14.0.2" 2020-07-14
OpenJDK Runtime Environment (build 14.0.2+12-46)
OpenJDK 64-Bit Server VM (build 14.0.2+12-46, mixed mode, sharing)

git version 2.31.1

node: v14.16.0

npm: 6.14.11

Docker version 20.10.5, build 55c4c88

docker-compose version 1.29.0, build 07737305

No change to package.json was detected. No package manager install will be executed.

- [x] Checking this box is mandatory (this is just to show you read everything)

**Comment by VergilSkye** [comment]
I had this same problem in a presentation about a project a week ago, and I was needing a quick live fix so i change the node to 14.16.1 and npm to  6.14.12 and solve my problem. I don't think this happen because _npm audit_, i think it's about the **LTS** version of node and npm in github actions.

**Comment by merlinofcha0s** [comment]
Indeed It's the 7.x.x who has a problem with the 6.x version of npm it's working.
